lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -278,125 +315,132 @@ def allocate_address(ec2, domain, reuse_existing_ip_allowed):
         if unassociated_addresses:
             return unassociated_addresses[0], False
 
-    return ec2.allocate_address(domain=domain), True
+    if not domain:
+        domain = 'standard'
+
+    response = ec2.allocate_address(Domain=domain)
+    changed = True
+
+    return response, changed
 
 
 def release_address(ec2, address, check_mode):
-    """ Release a previously allocated elastic IP address """
+    # Release a previously allocated elastic IP address
 
     # If we're in check mode, nothing else to do
-    if not check_mode:
-        if not address.release():
-            EIPException('release failed')
-
-    return {'changed': True}
+    if check_mode:
+        changed = False
+        response = None
+    else:
+        response = ec2.release_address(
+            AllocationId=address.get('AllocationId'))
+        changed = True
 
+    return response, changed
 
-def find_device(ec2, module, device_id, isinstance=True):
-    """ Attempt to find the EC2 instance and return it """
 
-    if isinstance:
-        try:
-            reservations = ec2.get_all_reservations(instance_ids=[device_id])
-        except boto.exception.EC2ResponseError as e:
-            module.fail_json(msg=str(e))
+def find_device(ec2, module, device_id, is_aninstance=True):
+    # Attempt to find the EC2 instance and return it
 
-        if len(reservations) == 1:
-            instances = reservations[0].instances
-            if len(instances) == 1:
+    if is_aninstance:
+        reservations = ec2.describe_instances(
+            InstanceIds=[device_id])
+        if reservations:
+            instances = next(iter(reservations.values()))
```

Comment:
I'm not sure how reliable this will be, given the structure of the return from boto3.  It would be safer to iterate over the reservations["Instances"] specifically.  Otherwise you maybe iterating over "Groups" instead.